nrrd/reader.py

```diff
@@ -420,25 +425,37 @@ def read_data(header, fh=None, filename=None):
         raise NRRDError('Size of the data does not equal the product of all the dimensions: {0}-{1}={2}'
                         .format(total_data_points, data.size, total_data_points - data.size))
 
-    # Eliminate need to reverse order of dimensions. NRRD's data layout is the same as what Numpy calls 'Fortran'
-    # order, where the first index is the one that changes fastest and last index changes slowest
-    data = np.reshape(data, tuple(header['sizes']), order='F')
+    # In the NRRD header, the fields are specified in Fortran order, i.e, the first index is the one that changes
+    # fastest and last index changes slowest. This needs to be taken into consideration since numpy uses C-order
+    # indexing.
+    
+    # The array shape from NRRD (x,y,z) needs to be reversed as numpy expects (z,y,x).
+    data = np.reshape(data, tuple(header['sizes'][::-1]))
+
+    # Transpose data to enable Fortran indexing if requested.
+    if index_order == 'F':
+        data = data.T
 
     return data
 
 
-def read(filename, custom_field_map=None):
+def read(filename, custom_field_map=None, index_order='F'):
     """Read a NRRD file and return the header and data
 
     See :ref:`user-guide:Reading NRRD files` for more information on reading NRRD files.
 
+    .. note::
+            Users should be aware that the `index_order` argument needs to be consistent between `nrrd.read` and `nrrd.write`. I.e., reading an array with `index_order='F'` will result in a transposed version of the original data and hence the writer needs to be aware of this.
```

Comment:
This is similar to the note you had in the read NRRD section. I think this would confuse the average user into thinking that the index order must match the one in the NRRD file. Maybe a note in `nrrd.write` would be better stating that the index order MUST match the index order associated with the data?

Reply:
As mentioned in the previous comment, I think it is worth repeating. I would rather be repetitive than risking making users frustrated :smile:.

Reply:
Yeah, I think that could be true too. However, as I mentioned above, I think this is a good thing to put into `nrrd.write`. We can take the note that was placed in the `Writing NRRD files` above.

Reply:
Added.